framework/base/src/types/interaction/tx_exec/tx_exec_upgrade.rs

```diff
@@ -79,6 +79,41 @@ where
     }
 }
 
+impl<Api, Payment, Gas, FromSourceValue, RH>
+    Tx<
+        TxScEnv<Api>,
+        (),
+        &ManagedAddress<Api>,
```

Comment:
Not like this. To generalize, make `To` generic, make it implement `TxToSpecified`, which has method `with_value_ref`